enhancements/sig-policy/89-operator-policy-kind/README.md

```diff
@@ -0,0 +1,557 @@
+also provide new controls over the upgrades and uninstallation of these operators, which would be
+difficult with current solutions.
+
+### Goals
+
+1. It should be easy to install an operator on a managed cluster with just one policy.
+2. It should be possible to specify what upgrades should be allowed performed on an operator.
+3. The policy status should report on the health of the operator, both the OLM resources like the
+   Subscription, as well as the associated Deployments.
+4. The policy should report when the CatalogSource for the operator is unhealthy. It should be
+   configurable whether this will cause the policy to become NonCompliant.
+5. What happens when the policy is deleted should be configurable: the operator might be deleted, or
+   the operator and its CRDs might be deleted, or nothing might be deleted. By default, only the
+   Subscription and CSVs will be deleted (matching the suggestion from OLM).
+
+### Non-Goals
+
+1. The installation of OLM on managed clusters will not be automated by this enhancement.
+2. Management of "Operands" will not be included in this new type - they should be defined in
+   separate `ConfigurationPolicies`, possibly with a
+   [dependency](../74-order-policy-execution/README.md) on the `OperatorPolicy`. ("Operand" here
+   refers to resources that would be created by a user, and managed by the operator, eg instances of
+   CRDs which are defined by the operator.)
+3. Creation of `CatalogSources` will not be included in the new type - they should be defined in
+   separate `ConfigurationPolicies`.
+
+## Proposal
+
+### OperatorPolicy Syntax
+
+```yaml
+apiVersion: policy.open-cluster-management.io/v1beta1
+kind: OperatorPolicy
+metadata:
+  name: my-operator
+spec:
+  remediationAction: enforce # or inform
+  severity: medium
+  operatorGroup: # optional
+    name: my-operator-group
+    namespace: own-namespace
+    target:
+      namespaces:
+        - foo
+        - bar
+    # selector: # One of selector or namespaces is allowed.
+    #   matchLabels: # or matchExpressions
+    #     mydomain.io/prod: "true"
+    serviceAccountName: my-og-account # optional
+  subscription:
+    config: {} # optional
+    channel: stable
+    name: my-operator
+    namespace: own-namespace
+    installPlanApproval: Automatic # may be overridden to Manual by other settings
+    source: my-catalog
+    sourceNamespace: my-catalog-namespace
+    startingCSV: my-operator.v0.1.0 # optional
+  allowedCSVs:
+    - my-operator.v0.1.1
+    - my-operator.v0.2.0
+  # - "*" # Allow all
+  removalBehavior:
+    operatorGroups: DeleteIfUnused
+    subscriptions: Delete
+    clusterServiceVersions: Delete
+    installPlans: Keep
+    customResourceDefinitions: Keep
+    apiServiceDefinitions: Keep
+  statusConfig:
+    catalogSourceUnhealthy: Condition
+    deploymentsUnavailable: NonCompliant
+    upgradesAvailable: Condition
+    upgradesProgressing: NonCompliant
+```
```

Comment:
How would we use this to ensure the Subscription's status has **status.state: AtLatestKnown** without specifying the exact CSV?

Reply:
Setting `upgradesAvailable: NonCompliant` would cause the policy to become non-compliant when the Subscription has `status.state: UpgradePending`.